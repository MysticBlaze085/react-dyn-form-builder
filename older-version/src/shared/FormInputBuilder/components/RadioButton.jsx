import React from 'react';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';

const RadioButton = ({ handler, formState, meta: { label, items, classes } }) => {
    const availableItems = items.map((item) => (
        <FormControlLabel
            key={item.key}
            control={<Radio value={item.key} label={item.value} {...handler('radio', item.key)} />}
            label={item.value}
        />
    ));
    return (
        <RadioGroup style={{ width: '100%' }} value={formState} className={classes}>
            <FormLabel component="legend">{label}</FormLabel>
            {availableItems}
        </RadioGroup>
    );
};

export default RadioButton;
