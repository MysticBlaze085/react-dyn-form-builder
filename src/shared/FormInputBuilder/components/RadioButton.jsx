import React from 'react';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';

const RadioButton = ({ handler, formState, meta: { label, options } }) => {
    const availableOptions = options.map((option) => (
        <FormControlLabel
            key={option.key}
            control={<Radio value={option.key} label={option.value} {...handler('radio', option.key)} />}
            label={option.value}
        />
    ));
    return (
        <RadioGroup value={formState}>
            <FormLabel component="legend">{label}</FormLabel>
            {availableOptions}
        </RadioGroup>
    );
};

export default RadioButton;
