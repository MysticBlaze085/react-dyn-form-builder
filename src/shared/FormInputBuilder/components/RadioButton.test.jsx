import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from './RadioButton';
const MOCK_DATA = {
    formState: 'other',
    meta: {
        label: 'Gender',
        items: [
            { key: 'male', value: 'Male' },
            { key: 'female', value: 'Female' },
            { key: 'other', value: 'other' },
        ],
        classes: 'col-6',
    },
    options: {},
    handler: () => {},
};
it('should render RadioButton componet', () => {
    expect(shallow(<RadioButton {...MOCK_DATA} />)).toMatchSnapshot();
});
