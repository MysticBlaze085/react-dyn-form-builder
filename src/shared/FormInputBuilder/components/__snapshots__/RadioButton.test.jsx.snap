// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RadioButton componet 1`] = `
<ForwardRef(RadioGroup)
  className="col-6"
  style={
    Object {
      "width": "100%",
    }
  }
  value="other"
>
  <WithStyles(ForwardRef(FormLabel))
    component="legend"
  >
    Gender
  </WithStyles(ForwardRef(FormLabel))>
  <WithStyles(ForwardRef(FormControlLabel))
    control={
      <ForwardRef(WithStyles)
        label="Male"
        value="male"
      />
    }
    key="male"
    label="Male"
  />
  <WithStyles(ForwardRef(FormControlLabel))
    control={
      <ForwardRef(WithStyles)
        label="Female"
        value="female"
      />
    }
    key="female"
    label="Female"
  />
  <WithStyles(ForwardRef(FormControlLabel))
    control={
      <ForwardRef(WithStyles)
        label="other"
        value="other"
      />
    }
    key="other"
    label="other"
  />
</ForwardRef(RadioGroup)>
`;
